docs/latex/Inhalt/03_Inhalt/03_clean_architecture.tex

```diff
@@ -0,0 +1,61 @@
+
+\begin{figure}[ht!]
+    \includegraphics[width=0.98\columnwidth]{../diagrams/application_uml.pdf}
+    \caption{Klassendiagramm der Application Layer}
+    \label{fig:class-diag-application}
+\end{figure}
+
+Sie Implementiert in die eingangs beschriebenen Anwendungsfälle gruppiert in den drei in \autoref{fig:class-diag-application} dargestellten Services \code{ProfileService}, \code{MatchingService} und \code{RecipeService}. Die einzigen Dependencies bestehen dabei zwischen den Serviceklassen und auf die beiden im Domaincode durch Interfaces definierten Repositories.
+
+\section{Schicht 1: Adapters}
+Die Adapterschicht wurde im Module \code{1-quickie-adapters} implementiert und erfüllt in der zugrundeliegenden Software zwei Aufgaben: Zum einen werden die Daten durch Mapper-Klassen von dem Format, welches die Services aus der Application Layer liefern, in das Format übersetzt, welches die Plugins benötigen und umgekehrt. Zum anderen regelt sie die Kommunikation zwischen den Plugins und der Application Layer. Um hierbei eine möglichst große Entkopplung zwischen den Plugins und Application Layer zu erreichen, geschieht diese Kommunikation eventbasiert.
+
+\begin{figure}[ht!]
+    \includegraphics[width=0.98\columnwidth]{../diagrams/adapter_uml.pdf}
+    \caption{Vereinfachter Ausschnitt des Klassendiagramms der Adapterschicht}
+    \label{fig:class-diag-adapter}
+\end{figure}
+
+\autoref{fig:class-diag-adapter} zeigt einen Ausschnitt des Klassendiagramms der Adapterschicht in vereinfachter Form. Die Adapterschicht besitzt vier Packages \code{event}, \code{profile}, \code{recipe} und \code{persistence}. Letzteres wurde hier im Sinne der Übersichtlichkeit zunächst ausgespart. 
+
+Das Package \code{event} enthält Pure Fabrication Code, welcher ausschließlich die eventbasierte Kommunikation betrifft. Die Klasse \code{Event} repräsentiert hierbei ein Event, welches von der Benutzeroberfläche gesendet wird und besitzt einen \code{EventType}. Die Klasse \code{EventAnswer} bildet das Resultat der Verarbeitung eines Events ab, welches an die Benutzeroberfläche zurückgeschickt wird. Die Daten, welche mittels der \code{EventAnswer} an die Benutzeroberfläche geschickt werden, werden in einem separaten Objekt gehalten. Hierbei handelt es sich je nach Form der Daten um eine Instanz einer Klasse, welche das Interface \code{EventAnswerData} implementiert. Die Logik, welche beim Auftreten eines bestimmten Events aufgerufen wird, befindet sich in Callback-Klassen. Zu jedem \code{EventType} existiert eine eine entsprechende Callback-Klasse, welche das Interface \code{EventCallback} implementiert.
+
+In den Packages \code{profile} und \code{recipe} werden zum einen die entsprechenden Callback-Klassen implementiert und zum anderen Mapper-Klassen, welche die von der Application Layer erhaltenen Daten in das von der Benutzeroberfläche benötigte Format überführen.
```

Comment:
Das klingt schon, als würden in dem Package Dinge vermischt werden. Möchten wir das aufdröseln?

Reply:
Ja, klingt sinnvoll, dann jeweils zwei Subpackages `mappers` und `callbacks`?